xc/xc7/yosys/utils.tcl

```diff
@@ -0,0 +1,22 @@
+# Update the CLKOUT[0-5]_PHASE parameter values.
+# Due to the fact that Yosys doesn't support floating parameter values
+# i.e. treats them as strings, the parameter values need to be multiplied by 1000
+# for the PLL registers to have correct values calculated during techmapping.
+proc update_pll_params {} {
+    set temp_file "[pid].tmp"
+    select t:PLLE2_ADV -list -write $temp_file
+    set fh [open $temp_file r]
+    foreach cell [gets $fh] {
+	for {set output 0} {$output < 6} {incr output} {
+	    set param_name "CLKOUT${output}_PHASE"
+	    set param_value [getparam $param_name t:PLLE2_ADV]
+	    if {$param_value > 0} {
+		set new_param_value [expr {$param_value * 1000}]
```

Comment:
That may still yield in floating point value but I'm not a TCL expert. I'd do `[expr int(round($param_value * 1000))]`

Reply:
Who is a TCL expert? :) Nice catch though, I will use your TCL snippet so that we actually only set integer values on this parameter.